### PR TITLE
Serialisation for Perk list

### PR DESCRIPTION
**Pull request by deibuji**
Implements serialisation functionality for #17.

Serialisation:
- Serialise list of perkIds using url safe base64 including CRC16 checksum (buildId)
- Deserialisation functionality
- Generate URL for buildId

Also includes some refactoring of the code:
- Created enum PerkType
- Created two sub classes of Perks: Killer and Survivor
- Refactored rolling
- New perk lists: selectedPerks and rollablePerks
- Use direct reference for imports
- Move classes into classes dir
- Formatting fixes (removal of trailing spaces and add newlines to ends of files)

### CODE REVIEW

File: lib/pages/perkPage.dart [hunk 1]
```diff
@@ -1,55 +1,91 @@
...
   }
 
   void _rollTileCallback(int index, BuildContext context) {
-    // Check if randomlySelectedPerks is less than or equal to perkList
-    if (perkList.length <= randomlySelectedPerks.length) {
-      String perkType = killerMode ? 'killer' : 'survivor';
-      String message = 'You cannot reroll with only 4 $perkType perks selected';
+    // Check if randomlySelectedPerks is less than or equal to rollablePerks
+    if (rollablePerks.length <= numPerksToSelect) {
+      String mode = describeEnum(perkMode);
```
Comment (Toumari) [inline lib/pages/perkPage.dart:87]: You have covered the condition of a user trying to roll a single tile when there are only 4 perks selected. However, you can now press the randomise button and nothing will happen ( i.e. no snackbar)  could the logic for this be added? )

File: lib/pages/perkPage.dart [hunk 2]
```diff
@@ -168,11 +204,13 @@ class _PerkPageState extends State<PerkPage> {
...
                         scale: 1.5,
                         child: Switch(
-                          value: killerMode,
+                          value: perkModeSwitch,
                           onChanged: (value) async {
                             setState(() {
-                              killerMode = value;
-                              setPerkListAndRandomise();
+                              perkModeSwitch = value;
+
```
Comment (Toumari) [inline lib/pages/perkPage.dart:211]: Blank line here that can be removed from what I can see?

**Review by Toumari — CHANGES_REQUESTED** [review]

File: lib/pages/perkPage.dart (hunk 2 above)
Comment (deibuji) [reply lib/pages/perkPage.dart:211]: There's only 1 blank line.

File: lib/pages/perkPage.dart (hunk 1 above)
Comment (deibuji) [reply lib/pages/perkPage.dart:87]: Applied

**Review by Toumari — APPROVED** [review]

**Pull request merged**